### PR TITLE
Feat/task delete options

### PR DESCRIPTION
**Pull request by docmerlin**
Allow delete for offset.

### CODE REVIEW

File: internal/ptrutil/ptr.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package ptrutil
```
Comment (mark-rushakoff) [inline internal/ptrutil/ptr.go]: I'm uncertain about this package for a few reasons.

- We've gotten this far without a new package for it.
- `pkg` is usually where we put these kinds of utilities. If we do keep it, we may as well let it be accessible from enterprise too.
- The `Ptr` suffix on the functions stutters with the package name.

Would you mind soliciting some feedback from the team about naming and placement for these, if we keep them?

File: task_test.go [hunk 2]
```diff
@@ -112,4 +115,29 @@ from(bucket:"x")
 			t.Fatalf("expected Cron to be \"\" but was %s", op.Cron)
 		}
 	})
+	t.Run("delete deletable option", func(t *testing.T) {
+		tu := &platform.TaskUpdate{}
+		tu.Options.Offset = ptrutil.DurationPtr(0)
+		newscript := `option task = {cron: "* * * * *", name: "foo"}
```
Comment (mark-rushakoff) [inline task_test.go]: Rename to `expScript` so it's more clear this is the expected final result?

File: task_test.go [hunk 3]
```diff
@@ -112,4 +115,29 @@ from(bucket:"x")
...
+	|> range(start: -1h)`
+		if err := tu.UpdateFlux(`option task = {cron: "* * * * *", name: "foo", offset: 10s} from(bucket:"x") |> range(start:-1h)`); err != nil {
+			t.Fatal(err)
+		}
+		op, err := options.FromScript(*tu.Flux)
+		if err != nil {
+			t.Error(err)
+		}
+		if op.Every != 0 {
+			t.Fatalf("expected every to be 10s but was %s", op.Every)
```
Comment (mark-rushakoff) [inline task_test.go]: s/10s/0/

File: internal/ptrutil/ptr.go (hunk 1 above)
Comment (docmerlin) [reply internal/ptrutil/ptr.go]: I'm totally cool with moving it to pkg.

File: task_test.go [hunk 4]
```diff
@@ -19,7 +21,7 @@ func TestOptionsMarshal(t *testing.T) {
 	if tu.Options.Every != 10*time.Second {
 		t.Fatalf("option.every not properly unmarshaled, expected 10s got %s", tu.Options.Every)
 	}
-	if tu.Options.Offset != time.Hour {
+	if tu.Options.Offset != nil && *tu.Options.Offset != time.Hour {
```
Comment (mark-rushakoff) [inline task_test.go]:
```suggestion
	if *tu.Options.Offset != time.Hour {
```

or

```suggestion
	if tu.Options.Offset == nil || *tu.Options.Offset != time.Hour {
```

File: pkg/pointer/pointer.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// Package pointer provides utilities for pointer handling that aren't avaliable in go.
+// Feel free to add more pointerification functions for more types as you need them.
+package pointer
+
+import "time"
+
+// Duration returns a pointer to its argument
```
Comment (mark-rushakoff) [inline pkg/pointer/pointer.go]: Please punctuate all the godoc in this file. https://blog.golang.org/godoc-documenting-go-code

**Comment by mark-rushakoff** [comment]
Can you add a new test to task/backend/servicetest, to exercise option deletion as it crosses through API/service boundaries?

- Create a new task with a Cron and Offset option
- Update the task to remove the Offset option, and change Cron to Every
- Retrieve the task again to ensure the options are now Every, without Cron or Offset

**Comment by mark-rushakoff** [comment]
The new test looks good -- I think it uncovered a legit JSON issue with the newly deletable options.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x82c095]

goroutine 1200 [running]:
testing.tRunner.func1(0xc000230800)
	/usr/local/go/src/testing/testing.go:792 +0x387
panic(0x1127100, 0x1f71290)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
encoding/json.(*encodeState).marshal.func1(0xc000515ad0)
	/usr/local/go/src/encoding/json/encode.go:301 +0x9a
panic(0x1127100, 0x1f71290)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/influxdata/influxdb.TaskUpdate.MarshalJSON(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2540be400, 0xc00153ce80, 0x0, 0x0, ...)
	/go/src/github.com/influxdata/influxdb/task.go:205 +0x1b5
encoding/json.marshalerEncoder(0xc000e16000, 0x123cd20, 0xc000daa000, 0x99, 0xc000510100)
	/usr/local/go/src/encoding/json/encode.go:453 +0xb7
encoding/json.(*encodeState).reflectValue(0xc000e16000, 0x123cd20, 0xc000daa000, 0x99, 0x2100100)
	/usr/local/go/src/encoding/json/encode.go:333 +0x82
encoding/json.(*encodeState).marshal(0xc000e16000, 0x123cd20, 0xc000daa000, 0x100, 0x0, 0x0)
	/usr/local/go/src/encoding/json/encode.go:305 +0xf4
encoding/json.Marshal(0x123cd20, 0xc000daa000, 0x123cd20, 0xc000daa000, 0xc0027d4280, 0x0, 0x0)
	/usr/local/go/src/encoding/json/encode.go:160 +0x52
github.com/influxdata/influxdb/http.TaskService.UpdateTask(0xc000922620, 0x16, 0xc000347020, 0x58, 0x0, 0x15ca9c0, 0xc000e05050, 0x390ce8780029001, 0x0, 0x0, ...)
	/go/src/github.com/influxdata/influxdb/http/task_service.go:1520 +0x15d
github.com/influxdata/influxdb/task/servicetest.testTaskOptionsUpdateFull(0xc000230800, 0xc00091c880)
	/go/src/github.com/influxdata/influxdb/task/servicetest/servicetest.go:385 +0x36d
github.com/influxdata/influxdb/task/servicetest.TestTaskService.func1.2(0xc000230800)
	/go/src/github.com/influxdata/influxdb/task/servicetest/servicetest.go:66 +0x4c
testing.tRunner(0xc000230800, 0xc0003faae0)
	/usr/local/go/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:878 +0x35c
```
